src/mesh_refiner.cpp

```diff
@@ -184,6 +184,14 @@ std::size_t Mesh_Refiner::mark_for_refinement (
       _cell_marker->set_value(index, true);
       marked_count++;
     }
+    if (Mesh_Refiner::_mesh->num_cells()+5*marked_count > Mesh_Refiner::max_elements)
```

Comment:
I added it to target. It does not over refine, but it try to refine all the time. Need to add a stop to say it can not find any tolerance for which you can refine